cache/FIFO_FH.h

```diff
@@ -606,12 +620,13 @@ bool FIFO_FHCache<TKey, TValue, THash>::find(TValue& ac,
     // we can re-check this and remove it if possible
 
     // Generally when we found the thing in frozen cache
-    if(m_fasthash->find(key, ac) && (ac != nullptr))
+    std::unique_lock<ListMutex> lockMap(m_mapMutex);
+    if(m_fasthash->find(key, ac) && (ac != nullptr) && (chunkMapper[key] > globalCounter))
```

Comment:
If I understand right, the keys are labelled increasingly from left to right: [1, 2, 3, 4, n] so I believe this is disabling the keys from left to right, but the lower priority segments are removed from right to left. I think we would need to start globalCounter at n or (reverse the chunk labelling to be decreasing).